CONTRIBUTING.md

```diff
@@ -0,0 +1,226 @@
+   
+7. Push to GitHub with:
+		
+```bash 
+git push origin feature-my-new-feature
+```  
+  
+ Finally, go to the web page of your fork of the Brainverse repo, and click   'Pull request' (PR) to send your changes to the maintainers for review. This   will send an email to the committers.  You can commit new changes to this branch and keep pushing to your remote -- github automatically adds them to your   previously opened PR.
+ 
+8. To obtain changes from main project repository `master` branch and merge to current branch:
+
+```bash
+git pull upstream master
+``` 
+    
+### Option 2
+
+0. Have a clone of our main [project repository](https://github.com/ReproNim/brainverse) as `origin` remote in your git:
+```bash
+git clone git://github.com/ReproNim/brainverse
+```
+1. Fork the [project repository](https://github.com/ReproNim/brainverse): click on the 'Fork'
+   button near the top right of the page.  This creates a copy of the code
+   base under your account on the GitHub server.
+
+2. Add your forked clone as a remote to the local clone you already have on your
+   local disk:
+
+          git remote add gh-YourLogin git@github.com:YourLogin/brainverse.git
+          git fetch gh-YourLogin
+
+    For this SSH URLs to work, you have to [generate an SSH keypair](https://help.github.com/articles/connecting-to-github-with-ssh/) in your computer and add it to your GitHub account.
+    
+    To ease addition of other github repositories as remotes, here is
+    a little bash function/script to add to your `~/.bashrc`:
+
+        ghremote () {
+                url="$1"
+                proj=${url##*/}
+                url_=${url%/*}
+                login=${url_##*/}
+                git remote add gh-$login $url
+                git fetch gh-$login
+        }
+
+    thus you could simply run:
+
+         ghremote git@github.com:YourLogin/brainverse.git
+
+    to add the above `gh-YourLogin` remote.  
+
+3. [Same as Option1 - Step 5] Create a branch (generally off the `origin/master`) to hold your changes.
+
+4. [Same as Option 1- Step 6] Work on this copy on your computer using Git to do the version control. 
+
+5. Push to GitHub with:
+
+          git push -u gh-YourLogin feature-my-new-feature
+
+   Finally, go to the web page of your fork of the Brainverse repo, and click
+   'Pull request' (PR) to send your changes to the maintainers for review. This
+   will send an email to the committers.  You can commit new changes to this branch and keep pushing to your remote -- github automatically adds them to your
+   previously opened PR.
+
+(If any of the above seems like magic to you, then look up the
+[Git documentation](http://git-scm.com/documentation) on the web.)
+
+Development environment
```

Comment:
i would suggest moving this more upfront. i.e. reorder the sections in terms of needed information. this part seems quite important - more so than git commands.

Reply:
OK, I will do that.